### PR TITLE
ref(replays): Ensure ReplayRecord has the most up-to-date field names

### PR DESCRIPTION
**Pull request by ryan953**
This PR kills `getEvent()` in ReplayReader in favor of `getReplay()`.

`getReplay()` returns an object of type `ReplayRecord`. To generate that type I was able to set a bunch of fields to be `null`.

The rest of the PR is renaming `.replayId` to be `.id` and the like.

### CODE REVIEW

File: static/app/views/replays/detail/eventMetaData.tsx [hunk 1]
```diff
@@ -19,16 +19,19 @@ type Props = {
...
   const errors = crumbs?.filter(crumb => crumb.type === 'error').length;
 
   return (
     <KeyMetrics>
       {replayRecord ? (
         <ProjectBadge
           project={
-            projects.find(p => p.id === replayRecord.projectId) || {
-              slug: replayRecord.projectSlug || '',
+            replayRecord.project || {
```
Comment (billyvg) [inline static/app/views/replays/detail/eventMetaData.tsx:32]: Yeah I think we'll want to use `useProjects` here instead and use the loading state from the hook.

Comment (ryan953) [reply static/app/views/replays/detail/eventMetaData.tsx:32]: fair enough. also it looks like this is the only spot that reads `.project`

File: static/app/views/replays/detail/eventMetaData.tsx [hunk 2]
```diff
@@ -19,20 +19,17 @@ type Props = {
 };
 
 function EventMetaData({crumbs, durationMs, replayRecord}: Props) {
-  const {projects} = useProjects();
+  const {
+    params: {replaySlug},
+  } = useRouteContext();
+  const [slug, id] = replaySlug.split(':');
```
Comment (billyvg) [inline static/app/views/replays/detail/eventMetaData.tsx:25]: I suppose this works - there's always the possibility of the project slug changing, but I don't see that being a concern for us.

**Review by billyvg — APPROVED** [review]

File: static/app/views/replays/detail/eventMetaData.tsx (hunk 2 above)
Comment (ryan953) [reply static/app/views/replays/detail/eventMetaData.tsx:25]: `useReplayData` uses the `projectSlug` straight out of the browsers-url to fetch all the data, just like it uses `orgSlug`. So i think if the slugs change the user will need to start over and navigate to the new url.

example, from useReplayData():
```typescript
  const [projectSlug, replayId] = replaySlug.split(':');
  ...
  const fetchReplay = useCallback(async () => {
    const response = await api.requestPromise(
      `/projects/${orgSlug}/${projectSlug}/replays/${replayId}/`
    );
    return response.data;
  }, [api, orgSlug, projectSlug, replayId]);
```

**Pull request merged**